### PR TITLE
Driver/Calibration API Integration

### PR DESCRIPTION
**Pull request by astaff**
This PR supersedes #361 and is a product of a rebase of c08a476 against `app-3-0-alpha`.

Note: #353 also captures interim review of the features and has code review feedback.

## overview

Partial work for alpha

Write a high level description of the pull request.

Calibration:
Calibration seems to be working for both pipettes. The functions to execute calibration actions can be found it `opentrons.util.calibration_functions`. 

Endpoints for functionality are:
`move_instrument_for_probing_prep(instrument, robot)`
and
`probe_instrument(instrument, robot)`

To do for Alpha:
- The persistance of offset adjustments between probing and runs still needs to be tested
- Tip length needs to be saved (a hacky way to do this could be to adjust the z-offset for the mount as is being done for `x` and `y` in the calibration functions.
- Adjust container files for containers in bradford assay protocol
- Implement `jog`

### CODE REVIEW

File: api/opentrons/api/calibration.py [hunk 1]
```diff
@@ -48,9 +48,14 @@ def move_to(self, instrument, obj):
         instrument._instrument.move_to(obj._container[0])
         self._set_state('ready')
 
-    def jog(self, instrument, coordinates):
+    def jog(self, instrument, distance, axis):
```
Comment (mcous) [inline api/opentrons/api/calibration.py]: Kinda feel like `instrument, axis, distance` is a more intuitive argument order, esp. if we want to have a default jog distance

File: api/opentrons/api/calibration.py [hunk 2]
```diff
@@ -48,9 +48,14 @@ def move_to(self, instrument, obj):
...
         self._set_state('moving')
-        instrument._instrument.jog(coordinates)
+        calibration_functions.jog_instrument(
```
Comment (mcous) [inline api/opentrons/api/calibration.py]: Same as above comment

File: api/opentrons/api/models.py [hunk 3]
```diff
@@ -20,7 +20,10 @@ def __init__(self, instrument, containers=None):
         self.id = id(instrument)
         self.name = instrument.name
         self.channels = instrument.channels
-        self.axis = instrument.axis
+        self.mount = instrument.mount
+        # TODO(artyom, 20171006): with axis removed from the instrument
+        # we still need to pass it to the UI for now
+        self.axis = 'a' if self.mount == 'right' else 'b'
```
Comment (mcous) [inline api/opentrons/api/models.py]: Didn't realize this was a what the API was doing to support UI. This comment is a TODO for myself to prioritize moving the app off `axis` and to `mount`

File: api/opentrons/data_storage/database_migration.py [hunk 4]
```diff
@@ -7,8 +7,41 @@
...
+
+    for well, opposite_offset in zip(sorted_wells, opposite_y_offsets):
+        x, y, z = well._coordinates
+        well._coordinates = Vector(x, opposite_offset, 0)
+        length, width = (well.properties['length'], well.properties['width'])
+        well.properties['length'] = width
+        well.properties['width'] = length
+
+
+def rotate_container_for_alpha(container):
```
Comment (mcous) [inline api/opentrons/data_storage/database_migration.py]: This function name 🤣

File: api/opentrons/trackers/pose_tracker.py [hunk 5]
```diff
@@ -178,3 +183,8 @@ def _object_moved(self, new_pos_msg: new_pos_msg):
 
     def relative_object_position(self, target_object, reference_object):
         return self[target_object].position - self[reference_object].position
+
+    def clear_all(self):
+        self._root_nodes = []
+        self._pose_dict = {}
+        self._node_dict = {}
```
Comment (mcous) [inline api/opentrons/trackers/pose_tracker.py]: Heads up: will this lack of newline at the end of the file fail linting?

**Comment by astaff** [comment]
Superseded by https://github.com/OpenTrons/opentrons/pull/374

**Pull request closed without merging**